recipes/gmp/all/conanfile.py

```diff
@@ -109,11 +109,10 @@ def generate(self):
                 "lt_cv_sys_global_symbol_pipe=cat",  # added to get further in shared MSVC build, but it gets stuck later
             ])
             tc.extra_cxxflags.append("-EHsc")
-            if (self.settings.compiler == "msvc" and Version(self.settings.compiler.version) >= "180") or \
-               (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "12"):
+            if check_min_vs(self, "180", raise_invalid=False):
                 tc.extra_cflags.append("-FS")
                 tc.extra_cxxflags.append("-FS")
-        env = tc.environment() # Environment must be captured *after* setting extra_cflags, etc. to pick up changes
+        env = tc.environment()
```

Comment:
Keep the comments pleases 🙏

Reply:
@prince-chrismc recipe is not the appropriate place for this kind of comment. If it's unclear, conan doc should be updated, or package templates if you want. Do you want to update hundreds of CCI recipes to explain how conan functions work?

Reply:
```suggestion
        env = tc.environment() # Environment must be captured *after* setting extra_cflags, etc. to pick up changes
```